tests/qdwh_test.py

```diff
@@ -85,38 +89,45 @@ def testQdwhUnconvergedAfterMaxNumberIterations(
           'm': m, 'n': n, 'log_cond': log_cond}
       for m, n in zip([8, 10, 20], [6, 10, 18])
       for log_cond in np.linspace(1, _MAX_LOG_CONDITION_NUM, 4)))
-  # TODO(tianjianlu): Fails on A100 GPU and TPU.
-  @jtu.skip_on_devices("gpu", "tpu")
+  # TODO(tianjianlu): Fails on A100 GPU.
+  @jtu.skip_on_devices("gpu")
   def testQdwhWithUpperTriangularInputAllOnes(self, m, n, log_cond):
     """Tests qdwh with upper triangular input of all ones."""
-    a = jnp.triu(jnp.ones((m, n)))
+    a = jnp.triu(jnp.ones((m, n))).astype(_QDWH_TEST_DTYPE)
     u, s, v = jnp.linalg.svd(a, full_matrices=False)
     cond = 10**log_cond
     s = jnp.linspace(cond, 1, min(m, n))
     a = (u * s) @ v
     is_symmetric = _check_symmetry(a)
     max_iterations = 10
-    actual_u, actual_h, _, _ = qdwh.qdwh(a, is_symmetric, max_iterations)
+
+    with jax.default_matmul_precision('float32'):
+      actual_u, actual_h, _, _ = qdwh.qdwh(a, is_symmetric, max_iterations)
     expected_u, expected_h = osp_linalg.polar(a)
 
     # Sets the test tolerance.
-    rtol = 1E6 * _POLAR_TEST_EPS
+    rtol = 1E6 * _QDWH_TEST_EPS
 
     with self.subTest('Test u.'):
-      self.assertAllClose(actual_u, expected_u, rtol=rtol)
+      relative_diff_u = _compute_relative_diff(actual_u, expected_u)
+      np.testing.assert_almost_equal(relative_diff_u, 1E-6, decimal=5)
 
     with self.subTest('Test h.'):
-      self.assertAllClose(actual_h, expected_h, rtol=rtol)
+      relative_diff_h = _compute_relative_diff(actual_h, expected_h)
+      np.testing.assert_almost_equal(relative_diff_h, 1E-6, decimal=5)
 
     with self.subTest('Test u.dot(h).'):
-      a_round_trip = actual_u.dot(actual_h)
-      self.assertAllClose(a_round_trip, a, rtol=rtol)
+      with jax.default_matmul_precision('float32'):
```

Comment:
You might find it cleaner to simply define a helper function:
```
_dot = partial(jnp.dot, precision="highest")
```
and use it everywhere in this test file, rather than sticking the scope in many places?